lib/parser_base.nit

```diff
@@ -72,15 +86,27 @@ class StringProcessor
 					break
 				end
 				if src[relpos] != i then
-					pos += 1
+					p += 1
 					fnd = false
 					break
 				end
 				relpos += 1
 			end
-			if fnd then return pos
+			if fnd then
+				_pos = p
+				return p
+			end
 		end
 	end
+
+	# Ignores any printable character until a whitespace is encountered
+	protected fun ignore_until_whitespace: Int do
+		while not src[pos].is_whitespace do pos += 1
+		return pos
+	end
+
+	# Returns the current location as a `Location` object
+	protected fun hot_location: Location do return new Location(line, line_offset)
```

Comment:
in the nit AST, hot_location is used to reefer the important part of a complex node. eg the name in a method in a call or the operator in an addition